ops/main.py

```diff
@@ -361,6 +361,15 @@ def _should_use_controller_storage(db_path: Path, meta: CharmMeta) -> bool:
         return False
 
 
+def _get_breaking_relation_id() -> Optional[int]:
+    """Returns the relation id of the currently breaking relation, if applicable."""
+    if not os.environ.get("JUJU_HOOK_NAME", "").endswith("-relation-broken"):
+        # Not the relation-broken event
+        return None
+
+    return int(os.environ.get("JUJU_RELATION_ID"))
```

Comment:
This is not actually correct, as `JUJU_RELATION_ID` has the format `relation-name:123`. Elsewhere in `main.py` this is parsed using `int(os.environ['JUJU_RELATION_ID'].split(':')[-1])`. So let's pull that one-liner out into `_parse_relation_id() -> int` and use that in both places.